### PR TITLE
[ONNX] Fix aten::new_zeros export

### PR DESCRIPTION
**Pull request by fxmarty**
Fixes export in https://github.com/pytorch/pytorch/issues/110597

cc @justinchuby

### CODE REVIEW

**Comment by justinchuby** [comment]
Could you enable the test in test_op_consistency and fix CLA? Thanks!

File: torch/onnx/symbolic_opset9.py
Comment (thiagocrepaldi) [inline torch/onnx/symbolic_opset9.py:3889]: This is using inter torchscript IR API, which is not ideal.

We have abstracted that away through [JitScalarType](https://github.com/pytorch/pytorch/blob/main/torch/onnx/_type_utils.py#L61) class that is called under the hood by `_try_get_scalar_type`

Why the current code is not good and how can we fix it without accessing private API from TorchScript here?

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset9.py
Comment (fxmarty) [reply torch/onnx/symbolic_opset9.py:3889]: I gave some detail about the issue with the current code here, https://github.com/pytorch/pytorch/issues/110597#issuecomment-1750331085, related to a (reason unidentified) change of behavior in torch.jit.trace.

I guess we could also modify https://github.com/pytorch/pytorch/blob/e9ebda29d87ce0916ab08c06ab26fd3766a870e5/torch/onnx/symbolic_helper.py#L78-L79 but it is unclear to me how so I modified directly the `new_zeros` function.

It seems that calling `_type_utils.JitScalarType.from_value(dtype)` on `(%15 : NoneType = prim::Constant(), scope: __main__.MyModel::)` raises:

```
torch.onnx.errors.SymbolicValueError: Cannot determine scalar type for this '<class 'torch.NoneType'>' instance and a default value was not provided.  [Caused by the value '15 defined in (%15 : NoneType = prim::Constant(), scope: __main__.MyModel::
)' (type 'NoneType') in the TorchScript graph. The containing node has kind 'prim::Constant'.] 
```

Comment (thiagocrepaldi) [reply torch/onnx/symbolic_opset9.py:3889]: Changing `_type_utils.JitScalarType.from_value(dtype)` would be a better solution. It is ok to use TorchScript API there because this is the abstraction layer between torchscript and onnx exporter

Comment (fxmarty) [reply torch/onnx/symbolic_opset9.py:3889]: Thank you. As you commented in the issue, it could be that the fix will be in torchscript anyway so I'll wait for an answer on that.

**Comment by thiagocrepaldi** [comment]
@fxmarty I am testing https://github.com/pytorch/pytorch/pull/110956  which pushes your fix to `from_value` while we don't hear from PyTorch

**Comment by thiagocrepaldi** [comment]
> Could you enable the test in test_op_consistency and fix CLA? Thanks!

@justinchuby adding `new_zeros` there raised those errors. Any idea on why?

```bash
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool SUBFAIL [0.0000s] [  0%]
onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool PASSED [0.1885s] [  0%]

=================================== FAILURES ===================================
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor(False), ())', kwargs='{}', opset=9, sample_num=0) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array(False)]. kw_inputs: {}. input names: [].
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor([[ True,  True, False,  True,  True],\n        [ True,  True,  True,  True,  True],\n        [False, False,  True, False, False],\n        [False, False, False,  True, False],\n        [ True,  True, False, False,  True]]), (2, 0))', kwargs='{}', opset=9, sample_num=1) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array([[ True,  True, False,  True,  True],
       [ True,  True,  True,  True,  True],
       [False, False,  True, False, False],
       [False, False, False,  True, False],
       [ True,  True, False, False,  True]]), array(2), array(0)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2'].
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor([], size=(0, 5, 0), dtype=torch.bool), (3, 2, 2))', kwargs='{}', opset=9, sample_num=2) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array([], shape=(0, 5, 0), dtype=bool), array(3), array(2), array(2)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2', 'onnx::Unsqueeze_3'].
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor([ True,  True,  True, False,  True]), (2, 3))', kwargs="{'dtype': torch.bool, 'device': 'cpu'}", opset=9, sample_num=3) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array([ True,  True,  True, False,  True]), array(2), array(3)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2'].
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor([False,  True, False,  True,  True]), (10,))', kwargs="{'dtype': torch.float64}", opset=9, sample_num=4) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array([False,  True, False,  True,  True]), array(10)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1'].
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor([False,  True,  True, False, False]), (1, 1, 12))', kwargs="{'device': 'cpu'}", opset=9, sample_num=5) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array([False,  True,  True, False, False]), array(1), array(1), array(12)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2', 'onnx::Unsqueeze_3'].
_ TestOnnxModelOutputConsistency_opset9CPU.test_output_match_new_zeros_cpu_bool (inputs='(tensor([ True, False,  True,  True,  True]), (2, 2, 2))', kwargs="{'dtype': torch.float64, 'device': 'cpu'}", opset=9, sample_num=6) _
Traceback (most recent call last):
  File "onnx/test_op_consistency.py", line 317, in test_output_match
    self.run_test(model, inputs, rtol=rtol, atol=atol)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 204, in run_test
    _run_test(model, tracing_remained_onnx_input_idx)
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 164, in _run_test
    return run_model_test(
  File "/var/lib/jenkins/workspace/test/onnx/onnx_test_common.py", line 84, in run_model_test
    return verification.verify(*args, options=options, **kwargs)
  File "<@beartype(torch.onnx.verification.verify) at 0x7f2ef0a3e160>", line 352, in verify
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 890, in verify
    _compare_onnx_pytorch_model(
  File "<@beartype(torch.onnx.verification._compare_onnx_pytorch_model) at 0x7f2ef0a271f0>", line 160, in _compare_onnx_pytorch_model
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 451, in _compare_onnx_pytorch_model
    compare_onnx_pytorch_model_with_input(input_args, input_kwargs)
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 443, in compare_onnx_pytorch_model_with_input
    onnx_outs = _run_onnx(onnx_session, onnx_inputs)
  File "<@beartype(torch.onnx.verification._run_onnx) at 0x7f2ef0a994c0>", line 16, in _run_onnx
  File "/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/onnx/verification.py", line 172, in _run_onnx
    raise ValueError(
ValueError: got too many positional inputs. inputs: [array([ True, False,  True,  True,  True]), array(2), array(2), array(2)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2', 'onnx::Unsqueeze_3'].
- generated xml file: /var/lib/jenkins/workspace/test/test-reports/python-pytest/onnx.test_op_consistency/onnx.test_op_consistency-38452ac968537a1e.xml -
=========================== short test summary info ============================
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array(False)]. kw_inputs: {}. input names: [].
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array([[ True,  True, False,  True,  True],
       [ True,  True,  True,  True,  True],
       [False, False,  True, False, False],
       [False, False, False,  True, False],
       [ True,  True, False, False,  True]]), array(2), array(0)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2'].
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array([], shape=(0, 5, 0), dtype=bool), array(3), array(2), array(2)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2', 'onnx::Unsqueeze_3'].
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array([ True,  True,  True, False,  True]), array(2), array(3)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2'].
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array([False,  True, False,  True,  True]), array(10)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1'].
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array([False,  True,  True, False, False]), array(1), array(1), array(12)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2', 'onnx::Unsqueeze_3'].
/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/__init__.py:635: UserWarning: torch.set_default_tensor_type() is deprecated as of PyTorch 2.1, please use torch.set_default_dtype() and torch.set_default_device() as alternatives. (Triggered internally at /var/lib/jenkins/workspace/torch/csrc/tensor/python_tensor.cpp:453.)
  _C._set_default_tensor_type(t)
SUBFAIL [0.0000s] onnx/test_op_consistency.py::TestOnnxModelOutputConsistency_opset9CPU::test_output_match_new_zeros_cpu_bool - ValueError: got too many positional inputs. inputs: [array([ True, False,  True,  True,  True]), array(2), array(2), array(2)]. kw_inputs: {}. input names: ['onnx::Unsqueeze_1', 'onnx::Unsqueeze_2', 'onnx::Unsqueeze_3'].
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 7 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
================== 7 failed, 1 passed, 62 deselected in 0.31s ==================
```

**Comment by justinchuby** [comment]
I am not very sure. It looks like the exported model was expected 3 inputs but the test provided 4.

**Comment by justinchuby** [comment]
Potentially, you may xfail the bool dtypes with an `xfail(..., dtypes=(torch.bool,))` entry to unblock this.

**Review by justinchuby — DISMISSED** [review]

File: torch/onnx/symbolic_opset9.py
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:3889]: dtype is an optional based on https://github.com/pytorch/pytorch/blob/80ea8784f392959f8d61e4cd37420819d5694bf1/aten/src/ATen/native/native_functions.yaml#L2343. So it is a valid condition we should account for imo.

Comment (thiagocrepaldi) [reply torch/onnx/symbolic_opset9.py:3889]: A regression has been acknowledged by PyTorch Core. By 10/13 we will have more information on how to address this upstream too

**Comment by thiagocrepaldi** [comment]
> I am not very sure. It looks like the exported model was expected 3 inputs but the test provided 4.

I think I know why. The tests for `aten::new_zeros` are exported without inputs, because all inputs are probably folded into graph constants. To fix that, the tests should run with `dynamic_axes` specified in the `torch.onnx.export` call

**Comment by thiagocrepaldi** [comment]
https://github.com/pytorch/pytorch/pull/110956 is being merged to fix this. @fxmarty has been added as co-author on the new PR

**Comment by fxmarty** [comment]
Thank you @thiagocrepaldi!

**Pull request closed without merging**
